tools/rtpsrelay/AssociationTable.h

```diff
@@ -11,23 +11,17 @@ namespace RtpsRelay {
 
 class AssociationTable {
 public:
-  explicit AssociationTable(const RelayAddresses& relay_addresses) :
-    relay_addresses_(relay_addresses)
-  {}
   void insert(const WriterEntry& entry,
-              RelayAddressesMap& relay_addresses_map);
+              GuidSet& guids);
```

Comment:
see comment in `RelayHandler.h` about using the return type instead of out-parameters.

Reply:
I would like to use an out parameter here because buried inside the qos index code is a loop that populates the map.  If I adopt this strategy everywhere, then I'm back to copying.